src/uikit/Input.js

```diff
@@ -1,4 +1,47 @@
+import React from 'react';
 import styled from 'react-emotion';
+import { withTheme } from 'emotion-theming';
+import Row from 'uikit/Row';
+import SearchIcon from 'react-icons/lib/fa/search';
+import FaTimesCircleO from 'react-icons/lib/fa/times-circle';
+
+// this this component should implement the default <input> type from react-dom with some additional
+// props for styling
+export const FilterInput = withTheme(
+  ({
+    theme,
+    LeftIcon = SearchIcon,
+    RightIcon = FaTimesCircleO,
+    className,
+    value,
+    disabled,
+    ...props
+  }) => {
+    const Wrapper = styled(Row)`
+      ${({ theme }) => theme.input};
+      color: ${({ theme }) => theme.greyScale7};
+      align-items: center;
+      .icon-left {
+        margin-right: 5px;
+      }
+      .icon-right {
+        cursor: pointer;
+      }
+      input {
+        border: none;
+        flex: 1;
+      }
+    `;
+    const clearInput = () => props.onChange({ target: { value: '' } });
```

Comment:
I'm worried about this, seems more natural to place a ref on `input` and call `inputRef.current.value = ''` so that the browser can take care of triggering events

Reply:
so this had to be a bit more involved than setting `inputRef.current.value`, but we have a pretty low-level browser-driven implementation now.